apps/ec.c

```diff
@@ -163,9 +163,7 @@ int ec_main(int argc, char **argv)
 
     if (!opt_cipher(ciphername, &enc))
         goto opthelp;
-    private = param_out || pubin || pubout ? 0 : 1;
-    if (text && !pubin)
-        private = 1;
+    private = (text && !pubin) || (!param_out && !pubin && !pubout);
```

Comment:
See above.

Reply:
Changed accordingly to `private = !pubin && (text || (!param_out && !pubout))`